scm/src/gmtb_scm_utils.F90

```diff
@@ -78,6 +78,24 @@ subroutine interpolate_to_grid_centers(n_input_levels, input_pres, input_field,
   !> @}
 end subroutine
 
+subroutine find_state_vertical_index_from_input(input_pres, pres_l, k_in, k_out)
```

Comment:
Wonder if this routine and `find_vertical_index_pressure` from `gmtb_scm_input.F90` could be consolidated into one?

Reply:
Yes, I think that you're right. The find_vertical_index_pressure was supposed to find the vertical index of a given pressure (or first below a given pressure). This one is finding the vertical index in one vertical array of pressure given a particular index in another vertical array of pressure, but it would be identical to just use the first one and give it input_pres(k_in) instead of the whole input_pres array and an index to use.

Reply:
@climbfuji See https://github.com/NCAR/gmtb-scm/pull/222/commits/f30731363b1d8bacf0ad673664c68f7b46edd48d for consolidated solution.